src/app/rpmostree-override-builtins.cxx

```diff
@@ -89,6 +91,47 @@ handle_override (RPMOSTreeSysroot  *sysroot_proxy,
   g_autoptr(GVariant) options = g_variant_ref_sink (g_variant_dict_end (&dict));
 
   g_autoptr(GVariant) previous_deployment = rpmostree_os_dup_default_deployment (os_proxy);
+//  g_autofree char *transaction_address = NULL;                                          
+//  if (!rpmostree_os_call_refresh_md_sync(os_proxy,&transaction_address,cancellable,error))
+//    return FALSE;
+  if (override_replace && opt_ex_pin_from_repos)
+    {
+      g_autoptr(GPtrArray) override_replace_final = NULL;
+      GRegex *is_http = g_regex_new("/^http:///", (GRegexCompileFlags)0, 
+                                    (GRegexMatchFlags)0, error);
+      GRegex *is_https = g_regex_new("/^https:///", (GRegexCompileFlags)0, 
+                                     (GRegexMatchFlags)0, error);
+      GRegex *ends_with_rpm = g_regex_new("/.rpm$/", (GRegexCompileFlags)0, 
+                                          (GRegexMatchFlags)0, error);
+      if(!is_http||!is_https||!ends_with_rpm)
+        return FALSE;
+
+      for (char **it = override_replace; it && *it; it++)
+        {
+          const char *pkg = *it;
+	  if(g_regex_match(is_http, (const char*) pkg, (GRegexCompileFlags)0, NULL) || 
+             g_regex_match(is_https, (const char*) pkg, (GRegexCompileFlags)0, NULL) || 
+             g_regex_match(ends_with_rpm, (const char*) pkg, (GRegexCompileFlags)0, NULL))
+            {
+              g_ptr_array_add(override_replace_final, pkg);
+                continue;                                                                    
+	    }
+          else
+            {                                                       
+              gchar ** urls = NULL; 
+              RPMOSTreeOSExperimental *repo = NULL; 
+              if (!rpmostree_osexperimental_call_search_sync (repo, pkg, &urls, 
```

Comment:
Sorry, my pseudo-code is wrong. We can call this only once after we coalesce all the queries to send.

Reply:
The first argument here should be `osexperimental_proxy`, which you can get from `rpmostree_load_os_proxies`.

Reply:
So you can change the `rpmostree_load_os_proxy` above at the top of this function to use `rpmostree_load_os_proxies` instead since we need both.